src/library/library.cpp

```diff
@@ -386,6 +394,10 @@ void Library::slotShowTrackModel(QAbstractItemModel* model) {
     emit(restoreSearch(trackModel->currentSearch()));
 }
 
+void Library::slotSaveViewState() {
```

Comment:
This slot should not be needed. As far as I remember you can directly connect signals to signals.